src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1183 @@
+
+    // Rename the overall class to [old class name][suffix].
+    String orgClassName = FilenameUtils.removeExtension(originalFile.getName());
+    new ClassRenamer().visit(compUnit, new String[] {orgClassName, "Minimized"});
+
+    // Write the compilation unit to the minimized file.
+    writeToFile(compUnit, minimizedFile);
+
+    // Compile the Java file and check that the exit value is 0.
+    if (compileJavaFile(minimizedFile, classPath, packageName, timeoutLimit) != 0) {
+      System.err.println("Error when compiling file " + filePath + ". Aborting.");
+      return false;
+    }
+
+    String runResult = runJavaFile(minimizedFile, classPath, packageName, timeoutLimit);
+
+    // expectedOutput is a map from method name to failure stack trace
+    // with line numbers removed.
+    Map<String, String> expectedOutput = normalizeJUnitOutput(runResult);
+
+    System.out.println("Minimizing: " + filePath);
+
+    // Minimize the Java test suite.
+    minimizeTestSuite(
+        compUnit,
+        packageName,
+        minimizedFile,
+        classPath,
+        expectedOutput,
+        timeoutLimit,
+        verboseOutput);
+
+    // Cleanup: simplify variable type names and sort the import statements.
+    compUnit =
+        simplifyVariableTypeNames(
+            compUnit, packageName, minimizedFile, classPath, expectedOutput, timeoutLimit);
+    sortImports(compUnit);
+
+    writeToFile(compUnit, minimizedFile);
+
+    System.out.println("Minimizing complete.\n");
+    System.out.println("Original file length: " + getFileLength(originalFile) + " lines.");
+    System.out.println("Minimized file length: " + getFileLength(minimizedFile) + " lines.");
+
+    return true;
+  }
+
+  /**
+   * Visit and minimize every method within a compilation unit.
+   *
+   * @param cu the compilation unit to minimize; will be modified by side effect
+   * @param packageName the package that the Java file is in
+   * @param file the Java file that is being minimized; is modified by side effect
+   * @param classpath classpath used to compile and run the Java file
+   * @param expectedOutput expected JUnit output when the Java file is compiled and run
+   * @param timeoutLimit number of seconds allowed for the whole test suite to run
+   * @param verboseOutput prints out information about minimization status if true
+   */
+  private static void minimizeTestSuite(
+      CompilationUnit cu,
+      String packageName,
+      File file,
+      String classpath,
+      Map<String, String> expectedOutput,
+      int timeoutLimit,
+      boolean verboseOutput) {
+    for (TypeDeclaration type : cu.getTypes()) {
+      for (BodyDeclaration member : type.getMembers()) {
+        if (member instanceof MethodDeclaration) {
+          MethodDeclaration method = (MethodDeclaration) member;
+
+          // Get method's annotations.
+          for (AnnotationExpr annotationExpr : method.getAnnotations()) {
+            if (annotationExpr.toString().equals("@Test")) {
```

Comment:
It's more natural to abstract this out into a routine, `isTestMethod`.
Then you can drop the for loop and the break statement, and the code will be easier to read.